eng/Generate.ps1

```diff
@@ -382,6 +383,22 @@ if (!$noBuild) {
     Invoke-TypeSpecSetup
 }
 
+[hashtable]$testPackagesToInstall = @{};
+$localTestNugetSourceFolder = Join-Path $repoRoot 'test/NugetPackages'
+foreach($key in $keys){
+    foreach($nugetPackageFilename in (Get-ChildItem -Path $localTestNugetSourceFolder | Select-Object -ExpandProperty Name)){
+        $nameForRegex = [Regex]::Escape($key);
```

Comment:
in which case will we need the escape here, can we just use `$key`? And `$name` could just be `$key`?

Reply:
when the key contains special char needs escape for regular expression like '.'. $key should equal to $name. adding var $name and $version just avoids using match group with magic number directly which is hard to read/understand, thx